src/Workspaces/Remote/ServiceHub/Services/SymbolFinder/RemoteSymbolFinderService.cs

```diff
@@ -227,19 +227,14 @@ public ValueTask OnDefinitionFoundAsync(SymbolGroup group, CancellationToken can
             }
 
             public async ValueTask OnReferencesFoundAsync(
-                IAsyncEnumerable<(SymbolGroup group, ISymbol symbol, ReferenceLocation location)> references,
+                ImmutableArray<(SymbolGroup group, ISymbol symbol, ReferenceLocation location)> references,
                 CancellationToken cancellationToken)
             {
-                using var _ = ArrayBuilder<(SerializableSymbolGroup, SerializableSymbolAndProjectId, SerializableReferenceLocation)>.GetInstance(out var result);
-                await foreach (var (group, symbol, location) in references)
-                {
-                    result.Add((
-                        SerializableSymbolGroup.Dehydrate(_solution, group, cancellationToken),
-                         SerializableSymbolAndProjectId.Dehydrate(_solution, symbol, cancellationToken),
-                         SerializableReferenceLocation.Dehydrate(location, cancellationToken)));
-                }
+                var dehydrated = references.SelectAsArray(
+                    r => (SerializableSymbolGroup.Dehydrate(_solution, r.group, cancellationToken),
+                          SerializableSymbolAndProjectId.Dehydrate(_solution, r.symbol, cancellationToken),
+                          SerializableReferenceLocation.Dehydrate(r.location, cancellationToken)));
```

Comment:
i'm not sure how i missed this.  this is the server-oop sending point.  When i switched this to IAsyncEnumerable, it meant that this was pulling on the entire stream, waiting for it to finish, before sending anything over to the client.  

switched to the ImmutableArray auto-chunking form. 

The semantics here are that as the server is sending this IA over, the ProducerConumser is still searching, and buffering up the results found after the last chunk was send to 'consumeItems'.  After this call to the client finishes, we'll get another consumeItems call with that buffer's content.  And that repeats until the producer is done.

Reply:
I've noticed a big lag on getting the initial reference the last couple days, just thought my machine was bugging out on me

Reply:
:'(